cmd/wasm-client/config.go

```diff
@@ -3,17 +3,41 @@
 
 package main
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+)
 
 type config struct {
 	DebugLevel string `long:"debuglevel" description:"Logging level for all subsystems {trace, debug, info, warn, error, critical} -- You may also specify <subsystem>=<level>,<subsystem2>=<level>,... to set the log level for individual subsystems -- Use show to list available subsystems"`
 	NameSpace  string `long:"namespace" description:"The name of the JS namespace in which all the call back functions should be registered"`
+
+	LocalPrivate string `long:"localprivate" description:"The static local private key to be used for creating the noise connection"`
+	RemotePublic string `long:"remotepublic" description:"The static remote public key to be used for creating the noise connection"`
+
+	OnLocalPrivCreate  string `long:"onlocalprivcreate" description:"The name of the js callback function that should be called once the local private key has been generated. This function is expected to persist the key so that it can be provided via a command line flag on a future connection"`
+	OnRemoteKeyReceive string `long:"onremotekeyreceive" description:"The name of the js callback function to be called once the client receives the remote static key. This function is expected to persist the kye so that it can be provided via a command line flag on a future connection"`
```

Comment:
Since we have a callback for the auth data now, should we go ahead and also expose that to the JS API? Then we could parse the macaroon and check if it's readonly on the terminal side.

Reply:
added!